Utilities/ITKv5Preparation/prefer-type-alias-over-typedef.sh

```diff
@@ -0,0 +1,35 @@
+cat > ./typdef_vim_script.vim << EOF
+:%s/typedef  *\(.*<\_s*.*[>,]\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+
+:%s/\*\(.*\) typedef\(s*\)\(.*\)\*/*\1 type alias\3*/ge
+:%s/\*\*\(.*\) typedef\(s*\)\(.*\)/**\1 type alias\3/ge
+:%s/\/\/\(.*\)typedefs/\/\/\1type alias/ge
+
+:%s/typedef  *\(.*<\_s*.*,\_s*.*,\_s*[^>]*>\)  *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/\/\/\(.*\)typedefs/\/\/\1type alias/ge
+:%s/typedef  *\(.*<\_s*.*[>,]\_s.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/\*\(.*\) typedef\(s*\)\(.*\)\*/*\1 type alias\3*/ge
+:%s/\*\*\(.*\) typedef\(s*\)\(.*\)/**\1 type alias\3/ge
+:%s/typedef  *\(.*[><,]*\_s*.*[><,]\_s.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[>,]\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*[>,]\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[><,]\_s*.*[^>]>\)\_s *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+
+:%s/typedef  *\(.*\) * \([a-zA-Z0-9]*\);/using \2 = \1;/g
+:%s/typedef  *\(.*<\_s*.*,\n[^>]*>\)  *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/typedef  *\(.*<\_s*.*,.*>\{-}>\)  *\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/using \2 = \1;/ge
+:%s/\(  *\)typedef  *\(.*<.*>\)\_s\(  *\)\([A-Za-z0-9][A-Za-z0-9]*\);\{-};/\1using \4 =\r\1    \2;/ge
+:%s/typedef support\(\.*\)/type alias support/ge
+
+:%s/  *;/;/ge
+:w
+:q
+EOF
+
+for ff in $(git grep -l "typedef" |head -n 1500); do
```

Comment:
This line was originally 
```
for ff in $(git grep -l "typedef" |fgrep -v ThirdParty |head -n 1500); do
```

I guess we do not wand to `fgrep` `ThirdParty` in this case.